src/node_state.rs

```diff
@@ -0,0 +1,341 @@
+        }
+
+        // Verify proof
+        let blob_detail = Self::verify_proof(&tx)?;
+
+        // hash payloads
+        let _blobs_hash = Self::extract_blobs_hash(&blob_detail)?;
+
+        // some verifications
+        // TODO commented until we have a real blobs_hash
+        //if blobs_hash != unsettled_tx.blobs_hash {
+        //    bail!(
+        //        "Proof blobs hash '{}' do not correspond to transaction blobs hash '{}'.",
+        //        blobs_hash,
+        //        unsettled_tx.blobs_hash
+        //    )
+        //}
+
+        self.save_blob_details(&tx, blob_detail)?;
+
+        // check if tx can be settled
+        let is_next_to_settle = self
+            .unsettled_transactions
+            .is_next_unsettled_tx(&tx.tx_hash, &tx.contract_name);
+
+        if is_next_to_settle && self.is_ready_for_settlement(&tx.tx_hash) {
+            // settle tx
+            self.settle_tx(&tx)?;
+        }
+
+        debug!("Done {:?}", self);
+
+        Ok(())
+    }
+
+    fn clear_timeouts(&mut self, height: &BlockHeight) {
+        let dropped = self.timeouts.drop(height);
+        for tx in dropped {
+            self.unsettled_transactions.remove(&tx);
+        }
+    }
+
+    fn save_blob_details(
+        &mut self,
+        tx: &ProofTransaction,
+        blob_detail: UnsettledBlobDetail,
+    ) -> Result<(), Error> {
+        let unsettled_tx = self
+            .unsettled_transactions
+            .get_mut(&tx.tx_hash)
+            .context("Tx is either settled or does not exists.")?;
+
+        // TODO: better not using "as usize"
+        unsettled_tx.blobs[tx.blob_index.0 as usize] = blob_detail;
+
+        Ok(())
+    }
+
+    fn is_ready_for_settlement(&self, tx_hash: &TxHash) -> bool {
+        let tx = match self.unsettled_transactions.get(tx_hash) {
+            Some(tx) => tx,
+            None => {
+                return false;
+            }
+        };
+
+        tx.blobs
+            .iter()
+            .all(|blob| blob.verification_status.is_success())
+    }
+
+    fn verify_proof(tx: &ProofTransaction) -> Result<UnsettledBlobDetail, Error> {
+        // TODO real implementation
+        Ok(UnsettledBlobDetail {
+            contract_name: tx.contract_name.clone(),
+            verification_status: VerificationStatus::Success(model::HyleOutput {
+                version: 1,
+                initial_state: StateDigest(vec![0, 1, 2, 3]),
+                next_state: StateDigest(vec![4, 5, 6]),
+                identity: Identity("test".to_string()),
+                tx_hash: tx.tx_hash.clone(),
+                index: tx.blob_index.clone(),
+                blobs: vec![],
+                success: true,
+            }),
+        })
+    }
+
+    fn extract_blobs_hash(blob: &UnsettledBlobDetail) -> Result<BlobsHash, Error> {
+        // TODO real implementation
+        match blob.verification_status {
+            VerificationStatus::Success(_) => Ok(BlobsHash::new("111")),
```

Comment:
```suggestion
            VerificationStatus::Success(hyle_output) => Ok(BlobsHash::new(hyle_output.blobs)),
```
Mais je pense qu'il faut changer la méthode new accordingly :smile: Et y'a ptetre des porblèmes de type qui vont se cacher à gauche à droite